airflow/jobs/scheduler_job.py

```diff
@@ -866,10 +866,26 @@ def _do_scheduling(self, session) -> int:
                 # Make sure we only sent this metric if we obtained the lock, otherwise we'll skew the
                 # metric, way down
                 timer.stop(send=True)
-            except OperationalError as e:
+            except (OperationalError, LockNotAvailable) as e:
```

Comment:
Thirdly: you aren't importing this from anywhere.

Reply:
yes not sure why its not converting to OperationalError. but i have tested mutliple times and behaviour is same.

Reply:
We also need to run without psycopg2 installed (i.e. if the user is on mysql then psycopg2 won't be around to import from)

Reply:
Yeah 'try/except ImportError` should be nicely handled here.